examples/mamba/README.md

```diff
@@ -0,0 +1,27 @@
+---
+library_name: nanotron
+---
+
+# Mamba
+
+Modeling code for Mamba to use with [Nanotron](https://github.com/huggingface/nanotron/)
+
+## 🚀 Quickstart
```

Comment:
Add instructions for compiling the kernel causal_conv1d_cuda or refer to https://github.com/state-spaces/mamba/issues/55#issuecomment-1858638484, so users don't forget to compile it. Because when I only ran pip install `causal-conv1d>=1.1.0,<1.2.0`, it raised an error

<img width="1204" alt="image" src="https://github.com/huggingface/nanotron/assets/22252984/f40855ff-b821-4a34-88aa-d46156606785">